### PR TITLE
KMA-24 Set up Styled Components

### PR DESCRIPTION
**Pull request by m-lo**
Convert to styled component (added comments about special cases, notably styling React Native Elements (vs. View / Text)
Added stories for Overlay
Fix ENV for React Native Storybook (include the env name in the babel config)
Rename a few variables
Add prestorybook to generate story exports

`export default process.env["REACT_NATIVE_STORYBOOK"] ? StorybookUIRoot : App;`
Not sure why process.env.REACT_NATIVE_STORYBOOK doesn't work for me

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -19,4 +19,4 @@ function App() {
   );
 }
 
-export default process.env.REACT_NATIVE_STORYBOOK ? StorybookUIRoot : App;
+export default process.env["REACT_NATIVE_STORYBOOK"] ? StorybookUIRoot : App;
```
Comment (m-lo) [inline App.js:22]: Not sure why dot access to env doesn't work for me

Comment (ericgmoon) [reply App.js:22]: It worked fine for me, but based on the response by Jack Zhang [here](https://stackoverflow.com/questions/33117227/setting-environment-variable-in-react-native), I have a feeling that the dot notation only works in certain shell environments.

Let's just add a global ignore for the `dot-notation` rule in `.eslintrc` in root.

**Comment by ericgmoon** [comment]
Could you please update the Snapshot tests after checking that they look fine using Storybooks?

File: babel.config.js [hunk 2]
```diff
@@ -5,7 +5,12 @@ module.exports = (api) => {
     presets: ["babel-preset-expo"],
     plugins: [
       "@babel/plugin-transform-react-jsx-source",
-      "transform-inline-environment-variables",
+      [
+        "transform-inline-environment-variables",
+        {
+          include: ["REACT_NATIVE_STORYBOOK"],
```
Comment (ericgmoon) [inline babel.config.js:11]: I don't think we need this? What is the purpose of this addition?

**Review by ericgmoon — CHANGES_REQUESTED** [review]

File: App.js (hunk 1 above)
Comment (m-lo) [reply App.js:22]: can ignore just that line

Comment (ericgmoon) [reply App.js]: This seems like a problem which would occur every time we use environment variables, so I think a global declaration would make more sense. Thoughts?

File: src/components/common/button/__snapshots__/test.js.snap [hunk 3]
```diff
@@ -114,8 +113,8 @@ exports[`renders correctly with icon on left 1`] = `
           "justifyContent": "center",
           "padding": 8,
           "paddingBottom": 16,
-          "paddingLeft": 12,
-          "paddingRight": 20,
```
Comment (ericgmoon) [inline src/components/common/button/__snapshots__/test.js.snap:118]: This doesn't seem right.

**Review by ericgmoon — APPROVED** [review]
Fixed a few bugs but otherwise looks great! :heavy_check_mark:

**Pull request merged**